pkg/bbgo/session.go

```diff
@@ -418,29 +420,35 @@ func (session *ExchangeSession) initSymbol(ctx context.Context, environ *Environ
 	for interval := range klineSubscriptions {
 		// avoid querying the last unclosed kline
 		endTime := environ.startTime
-		kLines, err := session.Exchange.QueryKLines(ctx, symbol, interval, types.KLineQueryOptions{
-			EndTime: &endTime,
-			Limit:   1000, // indicators need at least 100
-		})
-		if err != nil {
-			return err
-		}
+		var i int64
+		for i = 0; i < KLineLimit; i += 1000 {
+			var duration time.Duration = time.Duration(-i * int64(interval.Duration()))
+			e := endTime.Add(duration)
+
+			kLines, err := session.Exchange.QueryKLines(ctx, symbol, interval, types.KLineQueryOptions{
+				EndTime: &e,
+				Limit:   1000, // indicators need at least 100
+			})
+			if err != nil {
+				return err
+			}
 
-		if len(kLines) == 0 {
-			log.Warnf("no kline data for %s %s (end time <= %s)", symbol, interval, environ.startTime)
-			continue
-		}
+			if len(kLines) == 0 {
+				log.Warnf("no kline data for %s %s (end time <= %s)", symbol, interval, e)
+				continue
+			}
 
-		// update last prices by the given kline
-		lastKLine := kLines[len(kLines)-1]
-		if interval == types.Interval1m {
-			log.Infof("last kline %+v", lastKLine)
-			session.lastPrices[symbol] = lastKLine.Close
-		}
+			// update last prices by the given kline
+			lastKLine := kLines[len(kLines)-1]
+			if interval == types.Interval1m {
+				log.Infof("last kline %+v", lastKLine)
```

Comment:
this log is just to verify the lastPrice, so we can remove it now

Reply:
done